test/Drupal/Translatable/Content/FieldCollectionTranslatableTest.php

```diff
@@ -231,28 +231,153 @@ public function getTargetEntityFromStaticCache() {
+      $observerDrupal->expects($this->exactly(2))
         ->method('entityMetadataWrapper')
-        ->with($this->equalTo('field_collection_item'), $this->equalTo($willedFieldCollection))
-        ->willReturn($expectedTarget);
+        ->withConsecutive(array(
+          $this->equalTo($expectedHostEntityType),
+          $this->equalTo($expectedRawHost),
+        ), array(
+          $this->equalTo('field_collection_item')
+        ))
+        ->willReturnOnConsecutiveCalls($observerHost, $this->returnArgument(1));
+
+      // FieldCollectionTranslatable should attempt to access the target entity's
+      // host entity type, and set the host entity.
+      $observerTarget = $this->getMock('\FieldCollectionItemEntity', array(
+        'hostEntityType',
+        'setHostEntity'
+      ));
+      $observerTarget->expects($this->once())
+        ->method('hostEntityType')
+        ->willReturn($expectedHostEntityType);
+      $observerTarget->expects($this->once())
+        ->method('setHostEntity')
+        ->with($this->equalTo($expectedHostEntityType), $this->equalTo($expectedRawHost));
+
+      // We expect that the following properties on the target field collection
+      // will be removed.
+      $observerTarget->item_id = 'should be unset';
+      $observerTarget->revision_id = 'should also be unset';
+
+      // Set up the FieldCollectionTranslatable for testing and return the target.
+      $translatable = new MockFieldCollectionTranslatableForTargetEntityTest($mockWrapper, $observerDrupal);
+      $translatable->setExpectedTarget($observerTarget);
+      $translatable->setExpectedHost($expectedRawHost);
+      $actualTarget = $translatable->getTargetEntity($targetLang);
+
+      // Ensure that EntityTranslatableBase::getRawEntity() was called with the
+      // wrapped entity injected in the constructor.
+      $this->assertEquals($mockWrapper, $translatable->gotRawEntity);
+
+      // Ensure that FieldCollectionTranslatable::getHostEntity() was called with
+      // the target entity wrapper loaded from the prior getRawEntity() call.
+      $this->assertEquals($observerTarget, $translatable->gotHostEntity);
+
+      // Ensure the target entity returned is a FieldCollectionItemEntity.
+      $this->assertInstanceOf('\FieldCollectionItemEntity', $actualTarget);
+
+      // Ensure that the FieldCollectionItemEntity clone process set and unset all
+      // properties as expected.
+      $this->assertFalse(isset($actualTarget->item_id));
+      $this->assertFalse(isset($actualTarget->revision_id));
+      $this->assertEquals(TRUE, $actualTarget->is_new);
+    }
 
-      $observerWrapper = $this->getMock('\EntityDrupalWrapper', array('value'));
-      $observerWrapper->expects($this->once())
+    /**
+     * Tests the getTargetEntity method in the case that the host language does
+     * not match the target, and thus we need to create brand new field collection
```

Comment:
This is the place where we are having trouble. The test is correct, since the code IS trying to something different when the host language != the target language. This is where we have a listwrapper and the "host" is ending up looking at the source instead of the target.